src/IssueViz.UnitTests/Editor/QuickActions/QuickFixes/QuickFixSuggestedActionTests.cs

```diff
@@ -30,9 +34,68 @@ public class QuickFixSuggestedActionTests
         [TestMethod]
         public void DisplayName_ReturnsFixMessage()
         {
-            var testSubject = new QuickFixSuggestedAction("some fix");
+            var quickFixViz = new Mock<IQuickFixVisualization>();
+            quickFixViz.Setup(x => x.Fix.Message).Returns("some fix");
+
+            var testSubject = new QuickFixSuggestedAction(quickFixViz.Object, Mock.Of<ITextBuffer>());
+
+            testSubject.DisplayText.Should().Be("some fix");
+        }
+
+        [TestMethod]
+        public void Invoke_AppliesFixWithOneEdit()
+        {
+            var editVisualization = new Mock<IQuickFixEditVisualization>();
+            editVisualization.Setup(e => e.Edit.Text).Returns("edit");
+            
+            var quickFixViz = new Mock<IQuickFixVisualization>();            
+            quickFixViz.Setup(x => x.Fix.Message).Returns("some fix");
+            quickFixViz.Setup(x => x.EditVisualizations).Returns(new List<IQuickFixEditVisualization> { editVisualization.Object });
+
+
+            var textEdit = new Mock<ITextEdit>();
+
+            var textBuffer = new Mock<ITextBuffer>();
+            textBuffer.Setup(tb => tb.CreateEdit()).Returns(textEdit.Object);
+
+            var testSubject = new QuickFixSuggestedAction(quickFixViz.Object, textBuffer.Object);
+            testSubject.Invoke(new System.Threading.CancellationToken());
+
+            testSubject.DisplayText.Should().Be("some fix");
+            textBuffer.Verify(tb => tb.CreateEdit(), Times.Once(), "CreateEdit should be called once");
+            textEdit.Verify(tb => tb.Replace(It.IsAny<Span>(), It.IsAny<string>()), Times.Exactly(1), "Replace should be called one time");
```

Comment:
It's worth adding a stronger assertion. You can mock Edit.Span and Edit.Text and verify that Replace was called with the right arguments

Reply:
Because of sequence check, asserting the parameters became redundant.